tests/unit/overrides_test.yaml

```diff
@@ -0,0 +1,160 @@
+---
+suite: Override check
+templates:
+  # primary template files
+  - deployments.yaml
+  - jobs.yaml
+  - service-account.yaml
+
+  # included templates must also be listed
+  - configmaps_overrides.yaml
+  - configmaps_packs.yaml
+  - configmaps_rbac.yaml
+  - configmaps_st2-conf.yaml
+  - configmaps_st2-urls.yaml
+  - configmaps_st2web.yaml
+  - secrets_datastore_crypto_key.yaml
+  - secrets_ssh.yaml
+  - secrets_st2apikeys.yaml
+  - secrets_st2auth.yaml
+  - secrets_st2chatops.yaml
+
+tests:
+  - it: Jobs with overrides mounted
+    template: jobs.yaml
+    set:
+      st2:
+        overrides: #Enabling the override mounts in register-content job.
+          _global.yaml: |
+            ---
+            rules:
+              defaults:
+                enabled: false
+        rbac: { enabled: true } # enable rbac job
+        packs: { sensors: [] } # ensure only 1 sensor
+      jobs:
+        extra_hooks: &extra_hooks_jobs
+          - name: upgrade-warning
+            hook: pre-upgrade, pre-rollback
+            hook_weight: -5
+            command: ["st2", "run", "--tail", "custom_pack.warn_about_upgrade"]
+    release:
+      name: st2ha
+    asserts:
+      - hasDocuments:
+          count: 5
+
+      - contains: &override_volume
+          path: spec.template.spec.volumes
+          content:
+            name: st2-overrides-vol
+            configMap:
+              name: st2ha-st2-override-configs
+        documentIndex: 3 # register_content
+
+   
+      - contains: &override_mnt
+          path: spec.template.spec.containers[0].volumeMounts
+          content:
+            name: st2-overrides-vol
+            mountPath: /opt/stackstorm/overrides/
+        documentIndex: 3 # register_content
```

Comment:
```suggestion
      - contains: &overrides_volume
          path: spec.template.spec.volumes
          content:
            name: st2-overrides-vol
            configMap:
              name: st2ha-st2-overrides-configs
        documentIndex: 3 # register_content

   
      - contains: &overrides_mnt
          path: spec.template.spec.containers[0].volumeMounts
          content:
            name: st2-overrides-vol
            mountPath: /opt/stackstorm/overrides
        documentIndex: 3 # register_content
```

Reply:
And add
```
      - notContains: *overrides_volume
        documentIndex: 4 # extra_hooks job
      - notContains: *overrides_mnt
        documentIndex: 4 # extra_hooks job
```

Reply:
Added `documentIndex: 4` however set it as a contains as one of the requests further up was to add the mount to extra jobs :)

Reply:
oops. copy pasta - I meant to change that before submitting :facepalm: Thanks